api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2715,4 +2741,25 @@ service AuthService {
 
   // DeleteAllSAMLIdPServiceProviders removes all SAML IdP service providers.
   rpc DeleteAllSAMLIdPServiceProviders(google.protobuf.Empty) returns (google.protobuf.Empty);
+
+  // CreatePlugin creates a new plugin instance.
+  rpc CreatePlugin(CreatePluginRequest) returns (google.protobuf.Empty);
+
+  // GetPlugin returns a plugin instance by name.
+  rpc GetPlugin(types.ResourceWithSecretsRequest) returns (types.PluginV1);
+
+  // GetPlugins returns all plugin instances.
+  rpc GetPlugins(types.ResourcesWithSecretsRequest) returns (types.PluginListV1);
```

Comment:
This will be consumed by the UI. However, we expect a low amount of plugins in a cluster (~single digit). In fact, for the very first implementation, there will only be 1 type (Slack), and there will not be an option to connect multiple instances for one type. So the maximum amount can only be 1 plugin for now.

Please let me know if I should implement `ListPlugins` regardless for future-proofing.